source/slang/preprocessor.cpp

```diff
@@ -1622,10 +1626,21 @@ static void HandleIncludeDirective(PreprocessorDirectiveContext* context)
 
     // Push the new file onto our stack of input streams
     // TODO(tfoley): check if we have made our include stack too deep
+    auto sourceManager = context->preprocessor->getCompileRequest()->getSourceManager();
+
+    // See if this an already loaded source file
+    SourceFile* sourceFile = sourceManager->findSourceFile(foundPath);
```

Comment:
Should we be doing this check *before* we call through the user-provided file-system handler? It seems silly to ask them to find/load the file again, if we are then going to use our cached copy anyway.